astronomer/providers/google/cloud/xcom_backends/gcs.py

```diff
@@ -0,0 +1,67 @@
+import json
+import os
+import uuid
+from typing import Any, Union
+
+import pandas as pd
+from airflow.models.xcom import BaseXCom
+from airflow.providers.google.cloud.hooks.gcs import GCSHook
+
+
+class GCSXComBackend(BaseXCom):
+    """
+    Custom XCom persistence class extends base to support various datatypes.
+    Add the below line in the environment
+    AIRFLOW__CORE__XCOM_BACKEND=
+    astronomer.providers.google.cloud.xcom_backends.gcs.GCSXComBackend
+    """
+
+    PREFIX = os.getenv("PREFIX", "GCSXCOM_")
+    GCP_CONN_ID = os.getenv("CONNECTION_NAME", "google_cloud_default")
+    BUCKET_NAME = os.getenv("XCOM_BACKEND_BUCKET_NAME", "some_bucket_name")
+
+    @staticmethod
+    def write_and_upload_value(value: Any) -> str:
+        """Convert to string and upload to GCS"""
+        key_str = GCSXComBackend.PREFIX + str(uuid.uuid4())
+        hook = GCSHook(gcp_conn_id=GCSXComBackend.GCP_CONN_ID)
+        if isinstance(value, list):
+            value = str(value)
+        elif isinstance(value, dict):
+            value = json.dumps(value)
+        elif isinstance(value, pd.DataFrame):
+            value = value.to_json()
```

Comment:
I think you are missing a few other cases, no? What if you pass a custom obj? Shouldn't this also have an `else:` to handle the rest?

Reply:
```suggestion
        if isinstance(value, pd.DataFrame):
            value = value.to_json()
        else:
            value = json.dumps(value)
```
According to the [docs](https://docs.python.org/3/library/json.html#json.dumps) `json.dumps` can be of type `Any` 🤷 I am not saying it works for any type, but we could give it a try and test for which it does not. WDYT?

Reply:
Same thoughts as well. What would happen if users have pickling enabled (using our SDK for example)?

Reply:
fixed.